apps/rehash.c

```diff
@@ -427,9 +427,16 @@ static int do_dir(const char *dirname, enum Hash h)
                     while (bit_isset(idmask, nextid))
                         nextid++;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat"
```

Comment:
What if this file is compiled with a compiler other than gcc?

Reply:
I wonder if it would be better to rework the code to not use `%n`

Reply:
Then why provide the option?

Reply:
Yeah, maybe that would be better. Making the change isn't a big deal, but testing it... is there a test for it already? (I didn't look, I'm not on the computer.)

Reply:
> Then why provide the option?

It's very old. Probably from a time when you had to write your own format parser to be cross-platform.

Reply:
> I wonder if it would be better to rework the code to not use `%n`

Alternative fix in #22511.